test/parallel/test-promise-unhandled-warn.js

```diff
@@ -12,11 +13,25 @@ new Promise(() => {
 
 Promise.reject('test');
 
+function lookForMeInStackTrace() {
+  Promise.reject(new class ErrorLike {
+    constructor() {
+      Error.captureStackTrace(this);
+      this.message = 'ErrorLike';
+    }
+  }());
+}
+lookForMeInStackTrace();
+
 // Unhandled rejections trigger two warning per rejection. One is the rejection
 // reason and the other is a note where this warning is coming from.
-process.on('warning', common.mustCall(4));
+process.on('warning', common.mustCall((reason) => {
```

Comment:
What will happen if `--unhandled-rejections=throw` (which is the default nowadays)
does this code and test cover it?

Reply:
@ItamarGronich there is a separate test for `test-promise-unhandled-throw` though that's harder to test given it's process output.